pkg/scheduler/factory/factory.go

```diff
@@ -661,18 +660,21 @@ func (c *configFactory) updatePodInCache(oldObj, newObj interface{}) {
 					c.recorder.Eventf(updatedPod, v1.EventTypeNormal, "PodInPlaceResizeSuccessful", "Updated pod %s to resize resources", updatedPod.Name)
 				}
 			case schedulerapi.ResizeActionReschedule:
-				// Case 2. Node does not have capacity. Delete oldPod, let controller re-create pod.
-				oldPodName := oldPod.Name
-				c.recorder.Eventf(oldPod, v1.EventTypeNormal, "DeletePodForResizeReschedule", "Deleting pod %s to reschedule for resizing", oldPodName)
-				err := c.client.CoreV1().Pods(oldPod.Namespace).Delete(oldPod.Name, nil)
-				if err !=  nil {
+				// Case 2. Node does not have capacity. Delete pod, let controller re-create pod.
+				podName := newPod.Name
+				c.recorder.Eventf(newPod, v1.EventTypeNormal, "DeletePodForResizeReschedule", "Deleting pod %s to reschedule for resizing", podName)
+				deleteOptions := metav1.NewDeleteOptions(0)
```

Comment:
what are these for? deleteOptions and deleteOptions.Preconditions

Reply:
this is to ensure another pod created with the same name does not get unintentionally deleted. found this usage in another place in kubelet - i guess UID is guaranteed unique, name can collide, as unlikely as it might be.